uaclient/cli.py

```diff
@@ -839,21 +856,105 @@ def action_config_set(args, *, cfg, **kwargs):
         livepatch_status, _ = entitlement.application_status()
         if livepatch_status == ua_status.ApplicationStatus.ENABLED:
             configure_livepatch_proxy(**kwargs)
-    elif set_key in ("apt_http_proxy", "apt_https_proxy"):
+    elif set_key in apt_proxies_config_options:
         # setup_apt_proxy is destructive for unprovided values. Source complete
         # current config values from uaclient.conf before applying set_value.
-        protocol_type = set_key.split("_")[1]
+        proxy_scope = set_key.split("_")[0]
+        protocol_type_location = 1
+        if proxy_scope in ("global", "ua"):
+            protocol_type_location = 2
+        protocol_type = set_key.split("_")[protocol_type_location]
         if protocol_type == "http":
             validate_url = util.PROXY_VALIDATION_APT_HTTP_URL
         else:
             validate_url = util.PROXY_VALIDATION_APT_HTTPS_URL
         util.validate_proxy(protocol_type, set_value, validate_url)
-        kwargs = {
-            "http_proxy": cfg.apt_http_proxy,
-            "https_proxy": cfg.apt_https_proxy,
-        }
-        kwargs[set_key[4:]] = set_value
+
+        if proxy_scope == "apt":
+            event.info(ua_status.MESSAGE_WARNING_APT_PROXY_SETUP)
+            proxy_scope = "global"  # change the global_apt_* proxy config
+            set_key = "global_" + set_key
+            protocol_type_location = 2
+
+        # unset the current proxy setting
+        # either global or ua scoped
+        unset_current = False
+        if proxy_scope == "global":  # check if ua-scoped proxy set
+            unset_current = (
+                True
+                if getattr(cfg, "ua_apt_http_proxy")
+                or getattr(cfg, "ua_apt_https_proxy")
+                else False
+            )
+            if unset_current:
+                prev_val = "ua_apt_" + protocol_type + "_proxy"
+                event.info(
+                    ua_status.MESSAGE_WARNING_APT_PROXY_OVERWRITE.format(
+                        proxy=prev_val
+                    )
+                )
+        elif proxy_scope == "ua":  # check if global-scoped proxy set
+            unset_current = (
+                True
+                if getattr(cfg, "global_apt_http_proxy")
+                or getattr(cfg, "global_apt_https_proxy")
+                else False
+            )
```

Comment:
This pattern can be simplified a bit to:
```suggestion
            unset_current = bool(cfg.global_apt_http_proxy or cfg.global_apt_https_proxy)
```